src/base.ts

```diff
@@ -85,58 +40,78 @@ const getAuthorizationHeader = (accessToken = null) => {
   return { Authorization: `Bearer ${accessToken}` }
 }
 
-export interface RequestOptions {
-  isAuthorizedRequest?: boolean
+export const getHost = (host: string = null) => {
+  if (!apiClient.configuration.host) {
+    throw new Error(
+      'ApiClient not configured, please run: ApiClient.configure({ apiGatewayUrl: "your.api.gateway" }'
+    )
+  }
+
+  return host || apiClient.configuration.host
+}
+
+const buildHeaders = ({
+  headers = {},
+  recaptchaToken,
+  accessToken,
+  isAuthorizedRequest,
+}: RequestOptions): Record<string, string> => {
+  return {
+    "Content-Type": "application/json",
+    Accept: `application/vnd.carforyou.${apiClient.version}+json`,
+    ...(isAuthorizedRequest ? getAuthorizationHeader(accessToken) : {}),
+    ...(recaptchaToken ? { "Recaptcha-Token": recaptchaToken } : {}),
+    ...headers,
+  }
+}
+
+export interface ApiCallOptions extends Omit<RequestInit, "method" | "body"> {
+  recaptchaToken?: string
   accessToken?: string
-  method?: "GET" | "POST" | "PUT" | "PATCH" | "DELETE"
-  headers?: HeadersInit
+  headers?: Record<string, string>
```

Comment:
Just a side node: fech accepts the headers in different variations than just an object, we can go with objects only for our use case...